Content.Shared/Medical/Cryogenics/SharedCryoPodSystem.cs

```diff
@@ -120,18 +121,49 @@ protected void AddAlternativeVerbs(EntityUid uid, SharedCryoPodComponent cryoPod
             return;
 
         // Eject verb
-        if (cryoPodComponent.BodyContainer.ContainedEntity != null)
+        if (IsOccupied(cryoPodComponent))
         {
-            args.Verbs.Add(new AlternativeVerb
-            {
-                Text = Loc.GetString("cryo-pod-verb-noun-occupant"),
-                Category = VerbCategory.Eject,
-                Priority = 1, // Promote to top to make ejecting the ALT-click action
-                Act = () => TryEjectBody(uid, args.User, cryoPodComponent)
-            });
+            AlternativeVerb verb = new();
+            verb.Act = () => EjectBody(uid, cryoPodComponent);
+            verb.Category = VerbCategory.Eject;
+            verb.Text = Loc.GetString("cryo-pod-verb-noun-occupant");
+            verb.Priority = 1; // Promote to top to make ejecting the ALT-click action
+            args.Verbs.Add(verb);
+        }
+
+        // Self-insert verb
+        if (!IsOccupied(cryoPodComponent) &&
+            CanCryoPodInsert(uid, args.User, cryoPodComponent) &&
+            _blocker.CanMove(args.User))
+        {
+            AlternativeVerb verb = new();
+            verb.Act = () => InsertBody(uid, args.User, cryoPodComponent);
+            verb.Text = Loc.GetString("cryo-pod-verb-noun-occupant");
+            args.Verbs.Add(verb);
         }
     }
 
+    protected void AddInsertOtherVerb(EntityUid uid, SharedCryoPodComponent cryoPodComponent, GetVerbsEvent<InteractionVerb> args)
```

Comment:
while this technically works, this doesn't match the 'drag-drop' interaction pattern that SS14 (especially other medical entities) have - i'd say that this should be changed to match that, instead